### PR TITLE
docs: improve withTaskCancellationHandler docs

### PR DESCRIPTION
**Pull request by ktoso**
Docs improvements.

Also removes two places where we had deprecation warnings and can avoid them.

Please review @etcwilde @amartini51 

Resolves rdar://117284404

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: stdlib/public/Concurrency/TaskCancellation.swift [hunk 1]
```diff
@@ -22,27 +22,51 @@ import Swift
...
+/// performing any actual work once the operation gets to run.
+///
+/// The `operation` executes on the calling execution context and does not
+/// suspend or change execution context unless code contained within the closure
+/// does so. In other words, the potential suspension point of the
+/// `withTaskCancellationHandler` does not ever suspend by itself before
+/// executing the `operation`.
+///
+/// If cancellation occurs while the operation is running, the cancellation
+/// `handler` will execute *concurrently* with the `operation`.
```
Comment (amartini51) [inline stdlib/public/Concurrency/TaskCancellation.swift:60]:
```suggestion
/// The `operation` closure executes on the calling execution context, and doesn't
/// suspend or change execution context unless code contained within the closure
/// does so. In other words, the potential suspension point of the
/// `withTaskCancellationHandler(operation:onCancel:)` never suspends by itself before
/// executing the operation.
///
/// If cancellation occurs while the operation is running, the cancellation
/// handler executes *concurrently* with the operation.
```

Avoid future tense, use contractions, use code voice only when referring to a symbol in code.

**Review by amartini51 — APPROVED** [review]
Left some minor style corrections — these improvements look good to me.

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
```

<unknown>:0: note: No available targets are compatible with triple "riscv64-none-none-eabi"
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the crash backtrace.
Stack dump:
0.	<eof> parser at end of file
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x000000011060bfc7 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 39
```

trying again

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
Pending on https://github.com/apple/swift/pull/70057 to unbreak CI

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS

**Pull request merged**